contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -1017,6 +1018,111 @@ export const Bookmark = new Resource({
   getUserId: getUserIdFromStore,
 });
 
+export const CaptionFile = new Resource({
+  tableName: TABLE_NAMES.CAPTION_FILE,
+  urlName: 'captions',
+  idField: 'id',
+  indexFields: ['file_id', 'language'],
+  syncable: true,
+  getChannelId: getChannelFromChannelScope,
+
+  updateContentNodeChanged(id) {
+    return this.transaction({ mode: 'rw' }, CHANGES_TABLE, TABLE_NAMES.CONTENTNODE, () => {
+      ContentNode.table
+        .where({ channel_id: id })
+        .and(node => node.has_children == true)
+        .modify({ changed: true });
+    });
+  },
+
+  _add: Resource.prototype.add,
+  add(obj) {
+    return this._add(obj).then(() => { // id
+      const contentnodeId = this.getChannelId();
+      return this.updateContentNodeChanged(contentnodeId);
+    });
+  },
+
+  waitForCaptionCueGeneration(id) {
+    const observable = Dexie.liveQuery(() => {
+      return this.table
+        .where('id')
+        .equals(id)
+        .filter(f => !f[GENERATING])
+        .toArray();
+    });
+    return new Promise((resolve, reject) => {
+      const subscription = observable.subscribe({
+        next(result) {
+          if (result.length > 0 && result[0][GENERATING] === false) {
+            subscription.unsubscribe();
+            resolve(false);
+          }
+        },
+        error() {
+          subscription.unsubscribe();
+          reject();
+        },
+      });
+    });
+  },
+});
+
+export const CaptionCues = new Resource({
+  tableName: TABLE_NAMES.CAPTION_CUES,
+  urlName: 'captioncues',
+  idField: 'id',
+  indexFields: ['text', 'starttime', 'endtime', 'caption_file_id'],
```

Comment:
I'm thinking that `text` does not need to be indexed, because we aren't filtering or ordering by it.